added import-extensions-xlf gulp task

smartguest opened the pull request:
This PR automates a step in creating langpacks by copying valid ADS XLFs from 'export-xlfs/vscode-extensions' into the 'resources/xlf/en' folder.

Charles-Gagnon commented on build/gulpfile.vscode.js:
```diff
@@ -477,6 +477,47 @@ gulp.task(task.define(
+])
+
+// Copy ADS extension XLFs into English resource folder.
+gulp.task('import-extensions-xlfs', function () {
```
This seems to be a part of the "updating localization files" process you documented? It would be great if we could have a single task that runs all the necessary steps (up to the point of submitting a PR) together as well - is that something we're able to do?

Charles-Gagnon commented on build/gulpfile.vscode.js:
```diff
@@ -477,6 +477,47 @@ gulp.task(task.define(
+
+// Copy ADS extension XLFs into English resource folder.
+gulp.task('import-extensions-xlfs', function () {
+	return new Promise(function(resolve) {
```
Is the promise necessary here? I believe you should be able to make it

```
gulp.task('import-extensions-xlfs', function () {
		gulp.src(`../export-xlfs/vscode-extensions/*.xlf`)
			.pipe(extensionsFilter)
			.pipe(vfs.dest(`./resources/xlf/en`));
}
```

and it should behave the same

smartguest commented:
@Charles-Gagnon I've decided to modify vscode-translations-export with carbon edits, so now exports all extensions instead of a subset, I'm using that now instead. Is it okay to delete the vscode-translations-export folder after importing the xlfs (its outside the ADS directory, which might be dangerous)  or should that be an optional user step I should mention? It's only meant for cleanup of unnecessary files and does not affect the XLF creation and import process in any way.

Charles-Gagnon commented:
Why is it outside the ADS directory? Is that coming from the core VS code translation tasks? If possible putting it into the ADS directory would be preferable I think.

smartguest commented:
@Charles-Gagnon Yes its from the core task, I will add a carbon edit to change it to be inside the ads directory.

smartguest commented:
I have added tasks that copy over the ads langpacks md file to the vscode langpack, deletes the original ads langpack, then renames the vscode langpack to ads. Also moved the vscode-translations-export folder to an internal folder, and added a gulp task to delete it safely within ADS.

Charles-Gagnon commented on build/npm/import-language-xlfs.js:
```diff
@@ -0,0 +1,27 @@
+let gulp = require('gulp');
+let minimist = require('minimist');
+
+function update(options) {
```
Please add a comment to this file explaining what it's supposed to be doing.

Charles-Gagnon commented on build/npm/import-language-xlfs.js:
```diff
@@ -0,0 +1,27 @@
+function update(options) {
+	let location = options._;
+
+	gulp.src(`${location}/drop/loc/**/*.xlf`)
```
If this drop folder doesn't exist it would be helpful to print an error message informing the user what steps they need to do first

Charles-Gagnon commented on build/npm/import-language-xlfs.js:
```diff
@@ -0,0 +1,27 @@
+
+	gulp.src(`${location}/drop/loc/**/*.xlf`)
+		.pipe(gulp.dest(function (file) {
+			let firstSlashIndex =  file.relative.indexOf('\\');
```
Is this going to work cross-platform?

smartguest replied:
I used the relative file path for this and assuming the drop zip files are the same one, it should work, I'll test it on my mac as soon as I can.

Charles-Gagnon commented on build/npm/import-language-xlfs.js:
```diff
@@ -0,0 +1,27 @@
+		}));
+}
+
+if (path.basename(process.argv[1]) === 'import-language-xlfs.js') {
```
Any particular reason you made this a standalone script like this instead of a gulp task like you have for import-extensions-xlfs? Doesn't matter too much I guess but it's usually nice to keep similar pieces of code near each other.

smartguest replied:
Arguments for gulp task don't seem to exist from what I've been reading (this requires downloading extracting the drop folder), also the internal folder structure of the drop file is different from the actual structure of the resources/xlf folder. (I.E. In the drop folder, it is <language_id>/resources/xlf/en" instead of "resources/xlf/<language_id>"

Charles-Gagnon replied:
What do you mean "requires downloading extracting the drop folder"? And how does that have anything to do with this being a script?

Is it expected that the user has to pass in a folder location that may be outside the ADS folder? That definitely wasn't clear so if that's the case then I guess this is fine but you definitely should be documenting that. 

Where is this drop folder coming from and is there any way you could have the contents of this folder be created in the folder structure that we actually want?

smartguest replied:
You have to download this drop zip file containing the  XLFs from https://dev.azure.com/ms/azuredatastudio/_build?definitionId=453 and extracting the folder. I mentioned in my documentation that you needed to download a drop file from this pipeline, to do this. I'm not sure why the contents of the drop folder are created this way, but it appears to be related to something in LocProject.json. I'm going to investigate why this is the case.

Charles-Gagnon replied:
Ok, while you're investigating that could you remove this script from this PR for the time being? The rest of this looks fine so we can get that merged in - and then follow up with this later on once we figure out what the best way to approach that is.

smartguest replied:
Sure thing.
